### PR TITLE
Update module page

### PR DESCRIPTION
**Pull request by pascalgrimaud**
Following the discussion at https://groups.google.com/forum/#!topic/jhipster-dev/DogpMHZzf3w

The steps:
- review this page
- as soon as it's ok, I will do a new release of [generator-jhipster-module](https://github.com/jhipster/generator-jhipster-module)
- then, this PR can be merged

### CODE REVIEW

File: modules/creating_a_module.md [hunk 1]
```diff
@@ -35,20 +35,28 @@ A JHipster module:
...
+## Import the generator-jhipster
 
-A JHipster module uses the new "composability" feature from Yeoman, described at [http://yeoman.io/authoring/composability.html](http://yeoman.io/authoring/composability.html) to have access to JHipster's variables and functions.
+A JHipster module must import the generator-jhipster:
 
-For this, it composes with the "jhipster:modules" sub generator:
+```
+    const shelljs = require('shelljs');
+    const BaseGenerator = require('generator-jhipster/generators/generator-base');
+    const constants = require('generator-jhipster/generators/generator-constants');
```
Comment (deepu105) [inline modules/creating_a_module.md]: its better to name this as `jhipsterConstants`

File: modules/creating_a_module.md [hunk 2]
```diff
@@ -72,88 +80,51 @@ jhipsterFunc.registerModule(npmPackageName, hookFor, hookType[, callbackSubGener
...
-- `getExistingEntities`: get sorted list of entities according to changelog date in the current application
-- `isJhipsterVersionLessThan`: check if the JHipster version used to generate the project is less than a particular version
+## Available variables and functions
+
+### Variables from configuration:
 
+You have to use this function:
+
+```
+    _getConfig() {
```
Comment (deepu105) [inline modules/creating_a_module.md]: I think we can provide this function in generator base, so users can just use it

File: modules/creating_a_module.md [hunk 3]
```diff
@@ -72,88 +80,51 @@ jhipsterFunc.registerModule(npmPackageName, hookFor, hookType[, callbackSubGener
...
+            }
+        }
+        return false;
+    },
+```
+
+Then, you can access to configuration in `.yo-rc.json`:
+
+```
+    this.baseName = this._getConfig().baseName;
```
Comment (deepu105) [inline modules/creating_a_module.md]: this should advice assigning value to a local var
```
this.jhipsterAppConfig = this.getJhipsterAppConfig();
this.baseName = this.jhipsterAppConfig.baseName
```

File: modules/creating_a_module.md [hunk 4]
```diff
@@ -72,88 +80,51 @@ jhipsterFunc.registerModule(npmPackageName, hookFor, hookType[, callbackSubGener
...
+
+```
+    const javaDir = `${constants.SERVER_MAIN_SRC_DIR + this.packageFolder}/`;
+    const resourceDir = constants.SERVER_MAIN_RES_DIR;
+    const webappDir = constants.CLIENT_MAIN_SRC_DIR;
+```
+
+### Functions:
+
+You can use all functions in [generator-base](https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-base.js):
```
Comment (deepu105) [inline modules/creating_a_module.md]: We need to mention that only functions in `generato-base.js` and variables in `generator-constants.js` are part of public API and hence will follow semver. But other files like generator-base-private, utils etc will not follow semver and might break methods across minor versions

Comment (pascalgrimaud) [inline modules/creating_a_module.md]: updated above

File: modules/creating_a_module.md [hunk 5]
```diff
@@ -35,20 +35,28 @@ A JHipster module:
...
-        this.composeWith('jhipster:modules', {
-            options: {
-                jhipsterVar: jhipsterVar,
-                jhipsterFunc: jhipsterFunc
-            }
-        });
-    },
+Then, you can access to JHipster's variables and functions directly.
+
+**Note**: you don't need to use the composability described at [http://yeoman.io/authoring/composability.html](http://yeoman.io/authoring/composability.html) any more.
```
Comment (deepu105) [inline modules/creating_a_module.md]: I don't think we have to tell this, people can still use composability for other purposes.

File: modules/creating_a_module.md [hunk 6]
```diff
@@ -72,88 +80,41 @@ jhipsterFunc.registerModule(npmPackageName, hookFor, hookType[, callbackSubGener
...
+### Variables from configuration:
+
+You have to use this function:
+
+You can access to configuration in `.yo-rc.json`:
+
+```
+    this.jhipsterAppConfig = this.getJhipsterAppConfig();
+    this.baseName = this.jhipsterAppConfig.baseName;
+    this.baseName = this.jhipsterAppConfig.packageName;
```
Comment (deepu105) [inline modules/creating_a_module.md]: LHS is same for all variables, i think you forgot to change it

File: modules/creating_a_module.md [hunk 7]
```diff
@@ -72,88 +80,41 @@ jhipsterFunc.registerModule(npmPackageName, hookFor, hookType[, callbackSubGener
...
+### Functions:
+
+You can use all functions in [generator-base](https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-base.js):
+
+```
+    this.angularAppName = this.getAngularAppName(); // get the Angular application name.
+    this.printJHipsterLogo(); // to print the JHipster logo
+```
 
+**Note**: the functions in `generator-base.js` and variables in `generator-constants.js` are part of public API and hence will follow semver. But other files like `generator-base-private.js`, `utils.js` etc will not follow semver and might break methods across minor versions.
```
Comment (deepu105) [inline modules/creating_a_module.md]: The functions
semver versioning
break method signature

**Comment by pascalgrimaud** [comment]
I will try to find time tonight to do a new release of generator-jhipster-module, then this PR can be merged

**Comment by deepu105** [comment]
@pascalgrimaud but this change is valid once a new jhipster version is released (which is already done I guess) this change id not dependent on module generator I guess, please correct me if im wrong

**Comment by pascalgrimaud** [comment]
@deepu105 : indeed, I was wrong. This PR is not dependent on generator-jhipster-module

Only the next generator-jhipster-module depends on the new generator-jhipster 4.6.0 :) 

So this PR can be merged

**Pull request merged**